source/features/clean-conversation-headers.tsx

```diff
@@ -28,24 +28,30 @@ async function cleanPrHeader(byline: HTMLElement): Promise<void> {
 	const shouldHideAuthor
 		= pageDetect.isPRConversation()
 		&& !byline.closest('.gh-header-sticky') // #7802
-		&& $('.author', byline).textContent === (await elementReady('.TimelineItem .author'))!.textContent;
+		&& $([
+			'.author',
+			'a[data-hovercard-url]',
+		], byline).textContent === (await elementReady('.TimelineItem .author'))!.textContent;
 
 	if (shouldHideAuthor) {
 		byline.classList.add('rgh-clean-conversation-headers-hide-author');
 	}
 
-	const base = $('.commit-ref', byline);
+	const base = $([
+		'.commit-ref',
+		'[class^="BranchName"]',
+	], byline);
 	const baseBranchDropdown = $optional('.commit-ref-dropdown', byline);
 
 	// Shows on PRs: main [←] feature
 	const arrowIcon = <ArrowLeftIcon className="v-align-middle mx-1" />;
 	if (baseBranchDropdown) {
 		baseBranchDropdown.after(<span>{arrowIcon}</span>); // #5598
 	} else {
-		base.nextElementSibling!.replaceChildren(arrowIcon);
+		base.after(<span>{arrowIcon}</span>);
 	}
 
-	const baseBranch = base.title.split(':')[1];
+	const baseBranch = base.textContent.split(':')[1];
 	const wasDefaultBranch = pageDetect.isClosedPR() && baseBranch === 'master';
```

Comment:
Actually I don't konw what's it means here. But it caused below works wrong in PRCommitList.

- Merged PR (same author): https://github.com/sindresorhus/refined-github/pull/3402
- Merged PR (different author): https://github.com/parcel-bundler/parcel/pull/78
- Merged PR (different author + first published tag): https://github.com/sindresorhus/refined-github/pull/3227

Reply:
When the default branch is changed (as is common from `master` to `main`), every old PR appears as "merged into non-default" and therefore highlighted. Here we added an exception specifically for closed/old PRs so that the feature is "silent" in this very common scenario. 

It might be worth inlining this very comment in the code.

Reply:
> But it caused some problems

What problem is it?

Reply:
The baseBranch wich is default branch has been marked as `non-default-branch`

Reply:
Then I think the logic elsewhere is wrong. is `wasDefaultBranch` is true, then the result should be the opposite: `master` should be marked as default even if it currently isn't

Reply:
`pageDetect.isClosedPR` in PR conversation is `true`, but `false` in commit list.

<img width="224" alt="image" src="https://github.com/user-attachments/assets/deb74bab-042e-42a3-820d-97c01e8d8757">

<img width="224" alt="image" src="https://github.com/user-attachments/assets/ec185f72-2ca4-4535-8c7d-4c20564a1e8e">

Reply:
https://github.com/refined-github/github-url-detection/pull/200 @fregante

Reply:
Merged and published